Source/WebCore/html/HTMLCanvasElement.cpp

```diff
@@ -780,7 +780,8 @@ RefPtr<ImageData> HTMLCanvasElement::getImageData()
     if (document().settings().webAPIStatisticsEnabled())
         ResourceLoadObserver::shared().logCanvasRead(document());
 
-    RefPtr pixelBuffer = dynamicDowncast<ByteArrayPixelBuffer>(context->drawingBufferToPixelBuffer(GraphicsContextGL::FlipY::Yes));
+    RefPtr<ByteArrayPixelBuffer> pixelBuffer { };
```

Comment:
This isn't allocating anything, and you don't need to change this signature. The only thing needed would be allocation at the IPC level.